prime-dotnet-webapi/Services/Rules/AutomaticAdjudicationRules.cs

```diff
@@ -100,34 +104,45 @@ public override async Task<bool> ProcessRule(Enrollee enrollee)
                 {
                     record = await _collegeLicenceClient.GetCollegeRecordAsync(cert.License.Prefix, licenceNumber);
                 }
-                catch (PharmanetCollegeApiException)
+                catch (PharmanetCollegeApiException e)
                 {
                     enrollee.AddReasonToCurrentStatus(StatusReasonType.PharmanetError, licenceText);
+                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.License.Prefix, licenceNumber, e.Message);
                     passed = false;
                     continue;
                 }
                 if (record == null)
                 {
                     enrollee.AddReasonToCurrentStatus(StatusReasonType.NotInPharmanet, licenceText);
+                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.License.Prefix, licenceNumber, "Record not found in Pharmanet.");
                     passed = false;
                     continue;
                 }
 
                 if (!record.MatchesEnrolleeByName(enrollee))
                 {
-                    enrollee.AddReasonToCurrentStatus(StatusReasonType.NameDiscrepancy, $"{licenceText} returned \"{record.FirstName} {record.LastName}\".");
+                    string message = $"{licenceText} returned \"{record.FirstName} {record.LastName}\".";
+                    enrollee.AddReasonToCurrentStatus(StatusReasonType.NameDiscrepancy, message);
+                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.License.Prefix, licenceNumber, "(Name Discrepancy) " + message);
                     passed = false;
                 }
                 if (record.DateofBirth.Date != enrollee.DateOfBirth.Date)
                 {
-                    enrollee.AddReasonToCurrentStatus(StatusReasonType.BirthdateDiscrepancy, $"{licenceText} returned {record.DateofBirth:d MMM yyyy}");
+                    string message = $"{licenceText} returned {record.DateofBirth:d MMM yyyy}";
+                    enrollee.AddReasonToCurrentStatus(StatusReasonType.BirthdateDiscrepancy, message);
+                    await _businessEventService.CreatePharmanetApiCallEventAsync(enrollee.Id, cert.License.Prefix, licenceNumber, "(Birthdate Discrepancy) " + message);
                     passed = false;
```

Comment:
If someone enters the wrong License Number, but manage to enter a number for someone else, it is likely that both the name and birthdate will not match. This will create two business events, which I don't think we want because it will be misleading to the adjudicators. Also, this is now rather redundant with the status reason.

Reply:
I feel the intention of this business event was primarily to inform the adjudicators whether or not a call was made to the API. If we want to log the response from the API, it should be done in one location.

Reply:
My original implementation relied on `enrollee.AddReasonToCurrentStatus` and didn't have the `_businessEventService.CreatePharmanetApiCallEventAsync` within each of the conditional blocks.  But that didn't provide enough context in the business event message?  From Jira ticket:  "We should log what we sent to pharmanet, but not log the resulting dataset.  
Only record any mismatch in dataset and date/time of request."

For when both name and birthdate do not match, we can consolidate to a single business event re: something like enrollee likely entered license number of another practitioner?